Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -287,6 +287,32 @@ PhysicalParticleContainer::PhysicalParticleContainer (AmrCore* amr_core, int isp
     }
 #endif
 
+    // User-defined integer attributes
+    pp_species_name.queryarr("addIntegerAttributes", m_user_int_attribs);
+    int n_user_int_attribs = m_user_int_attribs.size();
+    std::vector< std::string > str_int_attrib_function;
+    str_int_attrib_function.resize(n_user_int_attribs);
+    m_user_int_attrib_parser.resize(n_user_int_attribs);
+    for (int i = 0; i < n_user_int_attribs; ++i) {
+        Store_parserString(pp_species_name, m_user_int_attribs[i]+"(x,y,z,ux,uy,uz,t)", str_int_attrib_function[i]);
+        m_user_int_attrib_parser[i] = std::make_unique<amrex::Parser>(
+                                            makeParser(str_int_attrib_function[i],{"x","y","z","ux","uy","uz","t"}));
+        AddIntComp(m_user_int_attribs[i]);
+    }
+
+    // User-defined real attributes
+    pp_species_name.queryarr("addRealAttributes", m_user_real_attribs);
+    int n_user_real_attribs = m_user_real_attribs.size();
+    std::vector< std::string > str_real_attrib_function;
+    str_real_attrib_function.resize(n_user_real_attribs);
+    m_user_real_attrib_parser.resize(n_user_real_attribs);
+    for (int i = 0; i < n_user_real_attribs; ++i) {
+        Store_parserString(pp_species_name, m_user_real_attribs[i]+"(x,y,z,ux,uy,uz,t)", str_real_attrib_function[i]);
```

Comment:
As above: let's use `.at()` over `[]` here

Reply:
thank you. done